src/main/java/com/atlassian/db/replica/internal/DatabaseMicroManager.java

```diff
@@ -0,0 +1,34 @@
+package com.atlassian.db.replica.internal;
+
+import com.atlassian.db.replica.api.reason.RouteDecision;
+import com.atlassian.db.replica.spi.Chief;
+
+import java.sql.Connection;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import static com.atlassian.db.replica.api.reason.Reason.WRITE_OPERATION;
+
+public class DatabaseMicroManager implements Chief {
```

Comment:
I like the micromanagement metaphor 😃 
But "database" seems too wide for a responsibility.

```suggestion
public class SqlQueryChief implements Chief {
```